testutils.py failure

While executing BLISS test suite, which partly rely on silx test suite for Flint, it happened that
`_outcome` was `None`:

```
self = <test_image_plot.TestImagePlot testMethod=test_display_image>

    def _currentTestSucceeded(self):
        if hasattr(self, '_outcome'):
            # For Python >= 3.4
            result = self.defaultTestResult()  # these 2 methods have no side effects
>           self._feedErrorsToResult(result, self._outcome.errors)
E           AttributeError: 'NoneType' object has no attribute 'errors'

../../miniconda3/envs/bliss/lib/python3.7/site-packages/silx/gui/utils/testutils.py:158: AttributeError
```
(maybe more for @vallsv ?)

--- a/silx/gui/utils/test/__init__.py
+++ b/silx/gui/utils/test/__init__.py
@@ -35,6 +35,7 @@ import unittest
 from . import test_async
 from . import test_image
 from . import test_qtutils
+from . import test_testutils
 from . import test
 
 
@@ -45,6 +46,7 @@ def suite():
     test_suite.addTest(test_async.suite())
     test_suite.addTest(test_image.suite())
     test_suite.addTest(test_qtutils.suite())
+    test_suite.addTest(test_testutils.suite())
     return test_suite
 
 

--- a/silx/gui/utils/test/test_testutils.py
+++ b/silx/gui/utils/test/test_testutils.py
@@ -29,6 +29,7 @@ __license__ = "MIT"
 __date__ = "16/01/2017"
 
 import unittest
+import sys
 
 from silx.gui import qt
 from ..testutils import TestCaseQt
@@ -37,6 +38,7 @@ from ..testutils import TestCaseQt
 class TestOutcome(unittest.TestCase):
     """Tests conversion of QImage to/from numpy array."""
 
+    @unittest.skipIf(sys.version_info.major <= 2, 'Python3 only')
     def testNoneOutcome(self):
         test = TestCaseQt()
         test._currentTestSucceeded()

--- a/silx/gui/utils/test/test_testutils.py
+++ b/silx/gui/utils/test/test_testutils.py
@@ -1,0 +1,53 @@
+# coding: utf-8
+# /*##########################################################################
+#
+# Copyright (c) 2017-2018 European Synchrotron Radiation Facility
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+# ###########################################################################*/
+"""Test of testutils module."""
+
+__authors__ = ["T. Vincent"]
+__license__ = "MIT"
+__date__ = "16/01/2017"
+
+import unittest
+
+from silx.gui import qt
+from ..testutils import TestCaseQt
+
+
+class TestOutcome(unittest.TestCase):
+    """Tests conversion of QImage to/from numpy array."""
+
+    def testNoneOutcome(self):
+        test = TestCaseQt()
+        test._currentTestSucceeded()
+
+
+def suite():
+    test_suite = unittest.TestSuite()
+    loader = unittest.defaultTestLoader.loadTestsFromTestCase
+    test_suite.addTest(loader(TestOutcome))
+    return test_suite
+
+
+if __name__ == '__main__':
+    unittest.main(defaultTest='suite')

--- a/silx/gui/utils/test/test_testutils.py
+++ b/silx/gui/utils/test/test_testutils.py
@@ -1,7 +1,7 @@
 # coding: utf-8
 # /*##########################################################################
 #
-# Copyright (c) 2017-2018 European Synchrotron Radiation Facility
+# Copyright (c) 2017-2019 European Synchrotron Radiation Facility
 #
 # Permission is hereby granted, free of charge, to any person obtaining a copy
 # of this software and associated documentation files (the "Software"), to deal

--- a/silx/gui/utils/testutils.py
+++ b/silx/gui/utils/testutils.py
@@ -155,7 +155,9 @@ class TestCaseQt(unittest.TestCase):
         if hasattr(self, '_outcome'):
             # For Python >= 3.4
             result = self.defaultTestResult()  # these 2 methods have no side effects
-            self._feedErrorsToResult(result, self._outcome.errors)
+            outcome = self._outcome
+            if outcome is not None:
+                self._feedErrorsToResult(result, outcome.errors)
         else:
             # For Python < 3.4
             result = getattr(self, '_outcomeForDoCleanups', self._resultForDoCleanups)

--- a/silx/gui/utils/testutils.py
+++ b/silx/gui/utils/testutils.py
@@ -155,9 +155,8 @@ class TestCaseQt(unittest.TestCase):
         if hasattr(self, '_outcome'):
             # For Python >= 3.4
             result = self.defaultTestResult()  # these 2 methods have no side effects
-            outcome = self._outcome
-            if outcome is not None:
-                self._feedErrorsToResult(result, outcome.errors)
+            if hasattr(self._outcome, 'errors'):
+                self._feedErrorsToResult(result, self._outcome.errors)
         else:
             # For Python < 3.4
             result = getattr(self, '_outcomeForDoCleanups', self._resultForDoCleanups)

